app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,345 @@
+			continue
+		}
+
+		encodedContent := base64.StdEncoding.EncodeToString(material.Content)
+		// create json message with the content
+		jsonMsg := fmt.Sprintf(`{"content": "%s"}`, encodedContent)
+		if err := i.sendWebhook(ctx, webhookURL, material.Type, json.RawMessage(jsonMsg), req.ChainloopMetadata); err != nil {
+			i.Logger.Errorw("failed to send SBOM webhook", "error", err, "type", material.Type)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// sendWebhook sends a webhook with the specified kind and payload
+func (i *Integration) sendWebhook(ctx context.Context, url, kind string, payload json.RawMessage, metadata *sdk.ChainloopMetadata) error {
+	payloadBytes, err := json.Marshal(webhookPayload{
+		Metadata: metadata,
+		Data:     payload,
+		Kind:     kind,
+	})
+	if err != nil {
+		i.Logger.Errorw("failed to marshal webhook payload", "error", err, "kind", kind)
+		return fmt.Errorf("marshalling webhook payload: %w", err)
+	}
+
+	reqHTTP, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewBuffer(payloadBytes))
+	if err != nil {
+		i.Logger.Errorw("failed to create HTTP request", "error", err, "url", url)
+		return fmt.Errorf("creating HTTP request: %w", err)
+	}
+	reqHTTP.Header.Set("Content-Type", "application/json")
+
+	resp, err := i.client.Do(reqHTTP)
+	if err != nil {
+		i.Logger.Errorw("failed to send HTTP request to webhook", "error", err, "url", url)
+		return fmt.Errorf("sending HTTP request to webhook: %w", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			i.Logger.Warnw("failed to close response body", "error", err)
+		}
+	}()
+
+	// Read response body for more detailed error messages
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		i.Logger.Warnw("failed to read response body", "error", err)
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		i.Logger.Errorw("webhook responded with non-success status code", "status", resp.StatusCode, "body", string(body))
+		return fmt.Errorf("webhook responded with status code %d: %s", resp.StatusCode, string(body))
+	}
+
+	return nil
+}
+
+// isBinary checks if the content is binary data
+func isBinary(content []byte) bool {
+	// Simple heuristic: check for null bytes
+	for _, b := range content {
+		if b == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// validateURL performs basic validation of the webhook URL
+func validateURL(webhookURL string) error {
+	parsedURL, err := url.ParseRequestURI(webhookURL)
+	if err != nil {
+		return fmt.Errorf("invalid URL format: %w", err)
+	}
+	if parsedURL.Scheme != "http" && parsedURL.Scheme != "https" {
+		return fmt.Errorf("unsupported URL scheme: %s", parsedURL.Scheme)
+	}
+	return nil
+}
+
+// testWebhookURL sends a POST request to ensure the webhook URL is reachable
+func testWebhookURL(client *http.Client, webhookURL string) error {
+	req, err := http.NewRequest(http.MethodPost, webhookURL, bytes.NewBuffer([]byte(`{"test": "test"}`)))
```

Comment:
I'd make the test message smth more specific so it can be identified in the other end. For example `[]byte(`{"test": "chainloop-plugin"}`)))`